crates/context/src/lib.rs

```diff
@@ -1050,12 +1050,22 @@ impl ContextManager {
 
         // println!("Propose in context members {:?}", members);
 
+        let proxy_contract: String = self
+            .config_client
+            .query::<ContextConfigEnv>(
+                context_config.protocol.as_ref().into(),
+                context_config.network.as_ref().into(),
+                context_config.contract.as_ref().into(),
+            )
+            .get_proxy_contract(context_id.rt().unwrap())
+            .await?;
```

Comment:
we should read this from the contract only once for the lifetime of the context, and store it in the context config of the local store

same as we do with `protocol`, `network`, `config_contract`, etc..

Reply:
-> Now in process of creating context / joining context it fetches proxy contract id , saves it and this are only times this is being called. 

In propose functions now it takes the argument from config